pages/product.page.ts

```diff
@@ -0,0 +1,40 @@
+import { expect, Locator } from '@playwright/test';
+import { BasePage } from './home/basePage';
+
+export class ProductPage extends BasePage {
+  readonly productName: Locator = this.page.getByTestId('product-name');
+  readonly unitPrice: Locator = this.page.getByTestId('unit-price');
+  readonly addToCartButton: Locator = this.page.getByTestId('add-to-cart');
+  readonly addToFavorites: Locator = this.page.getByTestId('add-to-favorites');
+
+  async expectProductDetails(
+    productName: string,
+    price: string
+  ): Promise<void> {
+    expect(this.page.url()).toContain('/product');
```

Comment:
```suggestion
    await expect(this.page).tohaveURL(/\/product$/);
```
To make test more stable

Reply:
fixed